Service/src/routes/portfolio.py

```diff
@@ -3,53 +3,74 @@
 import simplejson as json
 
 from routes import stock
+from auth import auth
+from request_validator import validator
+from request_validator.schemas import portfolio_schema
 from util.utility import Utility
 from util.error_map import errors
 
 portfolio_api = Blueprint('portfolio_api', __name__)
 
+DEFAULT_BUYPOWER = 10000
 
-@portfolio_api.route('/api/portfolio', methods=['POST'])
+@portfolio_api.route('/api/portfolios', methods=['POST'])
+@auth.login_required
+@validator.validate_body(portfolio_schema.fields)
 def post_portfolio():
    body = request.get_json()
-   try:
-      result = PortfolioSchema().load(body)
-   except ValidationError as err:
-      return make_response(json.dumps(err.messages), 400)
+   buyPower = body.get('buyPower') or DEFAULT_BUYPOWER
```

Comment:
Shouldn't we always require buying power to be submitted? If it has a default, I think it would be better to just have the frontend sending the default value.

Reply:
When do you see the endpoint being called without a specific buyPower value? In my opinion, it could cause unexpected behavior if the caller by mistake calls the endpoint without a buyPower but a value is assign regardless (and without the knowledge of the caller).

In the end it isn't a big deal as the frontend will always assign a buyPower (although that might be even more of a reason not to have a default).

Reply:
Yea I agree it's not a big deal because the frontend will assign a buyPower. However, I want the API to standalone. I could either throw an error or assign a default buy power. A user can later go ahead and delete their portfolio anyway so it's not a big deal to set a default.